client/packages.go

```diff
@@ -76,10 +81,97 @@ func (client *Client) Packages() (map[string]Package, error) {
 		return nil, fmt.Errorf("%s: response has no packages", errPrefix)
 	}
 
-	var packages map[string]Package
-	if err := json.Unmarshal(packagesJSON, &packages); err != nil {
+	var packageMap map[string]Package
+	if err := json.Unmarshal(packagesJSON, &packageMap); err != nil {
 		return nil, fmt.Errorf("%s: failed to unmarshal packages: %v", errPrefix, err)
 	}
 
+	var packages Packages
+	for _, pkg := range packageMap {
+		packages = append(packages, pkg)
+	}
+
 	return packages, nil
 }
+
+// IsInstalled returns true if the Package is currently installed
+func (p Package) IsInstalled() bool {
+	return p.Status == StatusInstalled || p.Status == StatusActive
+}
+
+// HasNameMatching returns true if the Package name matches the query
```

Comment:
Aha, thanks! Yeah, I think we should use regexp.QuoteMeta() or even 
```
func containsInsensitive(haystack, needle string) {
 return strings.Contains(strings.ToLower(haystack), strings.ToLower(needle))
}
``` 
to avoid that people start (ab)using the regexp "feature" :)